File: backend/components/metiers/administration_publique/utils/plugins/helpers/plugins_helper.test.py
# plugins_helper.test.py
# Tests unitaires Python pour plugins_helper
from .plugins_helper import is_valid_plugin_name

def test_is_valid_plugin_name():
    assert is_valid_plugin_name('plugin-1')
    assert is_valid_plugin_name('plugin_2')
    assert not is_valid_plugin_name('plugin!')
    assert not is_valid_plugin_name('')

File: backend/components/metiers/administration_publique/utils/plugins/helpers/plugins_helper.test.js
// plugins_helper.test.js
// Tests unitaires JS pour plugins_helper
const { isValidPluginName } = require('./plugins_helper');

describe('isValidPluginName', () => {
  it('valide un nom de plugin correct', () => {
    expect(isValidPluginName('plugin-1')).toBe(true);
    expect(isValidPluginName('plugin_2')).toBe(true);
  });
  it('rejette un nom de plugin invalide', () => {
    expect(isValidPluginName('plugin!')).toBe(false);
    expect(isValidPluginName('')).toBe(false);
  });
});
